src/hotspot/cpu/x86/x86.ad

```diff
@@ -8045,7 +8047,32 @@ instruct vprorate(vec dst, vec src, vec shift) %{
 %}
 
 #ifdef _LP64
-// ---------------------------------- Masked Block Copy ------------------------------------
+// ---------------------------------- Masked Operations ------------------------------------
+
+instruct vmask_cmp_node(rRegI dst, vec src1, vec src2, kReg mask, kReg ktmp1, kReg ktmp2, rFlagsReg cr) %{
+  match(Set dst (VectorCmpMasked src1 (Binary src2 mask)));
+  effect(TEMP_DEF dst, TEMP ktmp1, TEMP ktmp2, KILL cr);
+  format %{ "vector_mask_cmp $src1, $src2, $mask \t! vector mask comparison" %}
+  ins_encode %{
+    Label DONE;
+    const MachNode* src1_node = static_cast<const MachNode*>(this->in(this->operand_index($src1)));
+    const MachNode* src2_node = static_cast<const MachNode*>(this->in(this->operand_index($src2)));
+    assert(0 == Type::cmp(src1_node->bottom_type(), src2_node->bottom_type()), "");
+    int vector_len = vector_length_encoding(src1_node);
+    BasicType elmType =  src1_node->bottom_type()->is_vect()->element_basic_type();
+    __ knotql($ktmp2$$KRegister, $mask$$KRegister);
+    __ mov64($dst$$Register, -1L);
+    __ evpcmp(elmType, $ktmp1$$KRegister, $mask$$KRegister, $src1$$XMMRegister, $src2$$XMMRegister, Assembler::eq, vector_len);
+    __ kortestql($ktmp2$$KRegister, $ktmp1$$KRegister);
+    __ jccb(Assembler::carrySet, DONE);
+    __ kmovql($dst$$Register, $ktmp1$$KRegister);
+    __ notq($dst$$Register);
+    __ tzcntq($dst$$Register, $dst$$Register);
+    __ bind(DONE);
+  %}
+  ins_pipe( pipe_slow );
+%}
```

Comment:
The semantics of `VectorCmpMasked` seem to be defined nowhere. It doesn't much matter where you put the comment, perhaps on the definition of the node itself, but a reader shouldn't have to study the code to figure out what `VectorCmpMasked` is supposed to do.

Reply:
Hi @theRealAph , name VectorCmpMaskedNode is on the lines of other masked operations LoadVectorMaskedNode/StoreVectorMaskedNode.  Nomenclature used implicitly describe the semantics i.e. <Vector Operation>MaskedNode.  Appropriate comments appended.